database/migrations/2022_09_12_175257_rename_attributes_grants.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class RenameAttributesGrants extends Migration
+{
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		if (Schema::connection(null)->getConnection()->getDriverName() === 'sqlite') {
+			Schema::disableForeignKeyConstraints();
+		}
+		Schema::table('base_albums', function (Blueprint $table) {
+			$table->renameColumn('requires_link', 'is_link_required');
+		});
+		Schema::table('base_albums', function (Blueprint $table) {
+			$table->renameColumn('is_downloadable', 'grants_download');
+		});
+		Schema::table('base_albums', function (Blueprint $table) {
+			$table->renameColumn('grants_full_photo', 'grants_full_photo_access');
+		});
+		if (Schema::connection(null)->getConnection()->getDriverName() === 'sqlite') {
+			Schema::enableForeignKeyConstraints();
```

Comment:
Why is enabling/disabling foreign key constraints necessary?

I haven't checked it yet, but I assume the following happens. Renaming a column in SQLite is implemented by deleting the old table and creating a new table which is identical to the old one except for the renamed column. Although the renamed columns are not part of any foreign relations other columns of the affected table are and hence the foreign keys must be disabled as otherwise the table could not be deleted and re-created.

But this looks to me a lot like the same problem why we cannot delete columns from tables. Have you checked that all foreign constraints still exist afterwards and are still working?

Reply:
I hate sqlite.